lib/plugins/track.js

```diff
@@ -203,7 +203,8 @@ exports.resetUUID = function resetUUID (suggestedStorage, suggestedClientId) {
   this.client.track.uuid = clientId.replace(/\0/g, '')
 
   // Only save cookies if storage is enabled and expires is non-zero
-  if (storage) {
+  // and client is in signed mode
+  if (storage && this.inSignedMode()) {
```

Comment:
Maybe we're missing a mechanism to remove _td cookie? What if customer want to `resetUUID(undefined)` (in essential remove _td cookie). This new code will not allow it if they are in signed mode.

IMHO we can add a logic to remove _td cookie if `this.inSignedMode()` return `false`.